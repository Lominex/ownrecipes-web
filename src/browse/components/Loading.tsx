import { Col, Row } from 'react-bootstrap';

import CircularProgress from '../../common/components/CircularProgress';

const Loading: React.FC = () => (
  <Row>
    <Col xs={12}>
      <Row id='browse'>
        <div className='spinner'>
          <CircularProgress />
        </div>
      </Row>
    </Col>
  </Row>
);

export default Loading;
