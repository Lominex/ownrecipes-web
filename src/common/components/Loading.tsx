import Spinner from 'react-spinkit';

interface ILoadingProps {
  message?: string;
}

const Loading = ({ message }: ILoadingProps) => (
  <div className='spinner'>
    {message && <h3 className='no-results'>{ message }</h3>}
    <Spinner className='spinner-obj' name='circle' fadeIn='none' />
  </div>
);

export default Loading;
