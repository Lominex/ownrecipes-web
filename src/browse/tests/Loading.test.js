import Loading from '../components/Loading';
import renderer from 'react-test-renderer';

test('Loading component test', () => {
  const component = renderer.create(
    <Loading />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
