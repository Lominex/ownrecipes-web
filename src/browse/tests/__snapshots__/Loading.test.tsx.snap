// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component test 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <div
      className="row"
      id="browse"
    >
      <div
        className="spinner"
      >
        <div
          className="sk-spinner spinner-obj sk-circle"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
